src/main.rs

```diff
@@ -1,65 +1,66 @@
+use anyhow::{bail, ensure, Context, Result};
 use json_event_parser::{JsonEvent, JsonReader, JsonWriter};
 use std::env;
 use std::fs::File;
 use std::io::{BufRead, BufReader, Write};
 
-fn main() {
+fn main() -> Result<()> {
     env_logger::init();
 
-    let infile = match env::args().skip(1).next() {
-        Some(f) => f,
-        None => panic!("Missing required argument: path to StorageMarketDeals.json.zst"),
-    };
+    let infile = env::args()
+        .nth(1)
+        .context("Missing required argument: path to StorageMarketDeals.json.zst")?;
 
-    assert!(
+    ensure!(
         infile.ends_with(".json.zst"),
         "The StorageMarketDeals file must have .json.zst extension"
     );
-    let f = File::open(infile).expect("cannot open input file");
+
+    let f = File::open(&infile).context("cannot open input file")?;
     let decoder =
-        zstd::stream::Decoder::new(BufReader::new(f)).expect("cannot create zstd decoder");
+        zstd::stream::Decoder::new(BufReader::new(f)).context("cannot create zstd decoder")?;
     let mut reader = JsonReader::from_reader(BufReader::new(decoder));
 
     let mut buffer = Vec::new();
 
-    assert_eq!(
-        reader.read_event(&mut buffer).expect("cannot parse JSON"),
-        JsonEvent::StartObject,
+    ensure!(
+        reader.read_event(&mut buffer)? == JsonEvent::StartObject,
+        "read event not equal to JsonEvent::StartObject"
```

Comment:
Are you sure the new code is equivalent to the old one?

When `assert_eq!` fails, it prints a message describing the expected and the actual values.

Can `ensure!` do the same? It would have to parse the condition and extract the left-hand-side and right-hand-side values 🤔 

Example code:

```rs
    let x = 10;
    assert_eq!(x, 20)
```

Output

```
thread 'main' panicked at test/src/main.rs:3:5:
assertion `left == right` failed
  left: 10
 right: 20
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
I have tested `ensure!` macro with the equivalent code to the example you've provided:

```rust
let x = 10;
ensure!(x == 20);
```
And this is the result I got after running the program (without supplying the error message):

```
Error: Condition failed: `10 == 20` (10 vs 20)
```

It's also possible to supply the error message like so:

```rust
let x = 10;
ensure!(x == 20, "x is not equal to 20: x is {:?}", x);
```

Which in returns yields this result: 

```
Error: x is not equal to 20: x is 10
```

So I think in this case `ensure!` will fulfill the same use-case as `assert_eq!` but won't panic, rather it will return `Err` result.

Reply:
Here's a more concrete example that uses `JsonEvent` values:

```rust
let test_event = JsonEvent::EndObject;
ensure!(test_event == JsonEvent::StartObject);
```

Result without supplying error message:

```
Error: Condition failed: `test_event == JsonEvent::StartObject` (EndObject vs StartObject)
```

Reply:
Perfect, thank you for testing! My Rust experience is limited, I am learning here 🙈